CentralUnit/Communication/DirectSerialCommunication.cs

```diff
@@ -82,6 +82,11 @@ public void Write(byte[] input)
 
             try
             {
+                if (Log.IsDebugEnabled)
```

Comment:
This strikes me as weird. Does the logging library show debug messages when it is not enabled? This seems like an if the logger should be doing internally

Reply:
This is a simple boolean check which is faster than building a string and then throwing it away.